feature: clients info

LeonelCeballos18 opened the pull request:
## What kind of PR is this?
- [x] Feature

## Description
-add `customer dto` and `customer get response dto` and added in `index dto`
-add `get` and `get by id` methods in customer service and customer controller

## Ticket id
https://barbershop-app.atlassian.net/jira/software/projects/SCRUM/boards/1?selectedIssue=SCRUM-65

## Useful resources
<img width="1022" alt="Screenshot 2025-04-09 at 13 17 46" src="https://github.com/user-attachments/assets/5e3034c1-cf57-448b-97c5-24215df67a1e" />

LeonelCeballos18 committed:
feature: clients info

-add `customer dto` and `customer get response dto` and added in `index dto`
-add `get` and `get by id` methods in customer service and customer controller

cmagallon1 commented on src/services/customer.service.ts:
```diff
@@ -12,6 +12,65 @@ export default class CustomerService {
+      },
+    });
+
+    const users = customers.map((customer) => {
```
The ticket description says that you should include the ability to execute pagination if there are many clients. 

According to what's currently in the code, you're not supporting pagination; that process for now needs to be done in the front end if the situation arise. is that okay? otherwise I'd say that you need to add logic to paginate the results based on query params

LeonelCeballos18 replied:
The pagination is managed on frontend, we agree to do it in that way, but we never change the description of the ticket

cmagallon1 replied:
I see, all good then

LeonelCeballos18 replied:
fine

cmagallon1 commented on src/services/customer.service.ts:
```diff
@@ -12,6 +12,65 @@ export default class CustomerService {
+    });
+
+    const users = customers.map((customer) => {
+      const customerResponse = new CustomerResponseDto();
```
💭  this logic is duplicated in the getById method, you can abstract this into its own method and call it in both endpoints

cmagallon1 replied:
I'm referring the for an individual customer inside the map method 👀

LeonelCeballos18 replied:
fine, working on it

LeonelCeballos18 committed:
chore: Solving comments

-add `customerMapping` method to reuse customer mapping function in `get` and `getById` methods
